backend/resolver/aws/aws.go

```diff
@@ -136,3 +150,49 @@ func (r *res) Search(ctx context.Context, typeURL, query string, limit uint32) (
 		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("cannot search for type '%s'", typeURL))
 	}
 }
+
+func (r *res) Autocomplete(ctx context.Context, typeURL, search string, limit uint64) ([]*resolverv1.AutocompleteResponse_AutocompleteResult, error) {
+	if r.topology == nil {
+		return nil, fmt.Errorf("to use the autocomplete api you must first setup the topology service")
+	}
+
+	// TODO (mcutalo): Before implementing another resource to support autocomplete
+	// consider abstracting this into the topology service.
+	searchRequest := &topologyv1.SearchRequest{
+		PageToken: "0",
+		Limit:     resolver.DefaultAutocompleteLimit,
```

Comment:
should we check the limit before using the default limit? or maybe you are checking somewhere else and im misunderstanding the flow.